src/platform/packages/shared/kbn-std/src/process_in_chunks/process_in_chunks.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { Observable, OperatorFunction, from, lastValueFrom, mergeMap, reduce } from 'rxjs';
+import deepMerge from 'deepmerge';
+
+type CallbackFn<TResult> = (chunk: string[], id: number) => Promise<TResult>;
+
+const MAX_HTTP_LINE_LENGTH = 4096;
+// Apply an 80% threshold to the http line max length to guarantee enough space for url and potentially other parameters.
+// This value might need to vary as it's an estimate of how much we can reserve for the chunked list length.
+export const MAX_HTTP_LINE_CHUNK_SIZE = MAX_HTTP_LINE_LENGTH * 0.75; // 4096 *0.75 === 3072 characters, as 1 char = 1 byte
+
+/**
+ * This process takes a list of strings (for this use case, we'll pass it a list of data streams), and does the following steps:
+ * 1. Create chunks from the original list. Each chunk will contain as many items until their summed length hits the limit.
+ * 2. Provide each chunk in parallel to the chunkExecutor callback and resolve the result, which for our use case performs HTTP requests for data stream stats.
+ * 3. Deep merge the result of each response into the same data structure, which is defined by the first item in the list.
+ * 4. Once the observable used for these steps complete, we get the last value and returns it.
+ */
+export const processInChunks = <TResult>(
```

Comment:
Eh, this seems a little overengineered...?

Reply:
I didn't look that much at the implementation since it was already in place and tested, this was only moved/renamed and I preferred to initially move on with this since the bug is in production.

However, I simplified the implementation to use native JS features, let me know if it looks better, or if you have any alternative to suggest: [refactor(std): simplify implementation processInChunks](https://github.com/elastic/kibana/pull/222969/commits/098914a01a809c257e06393b6aac0aca3ed2baca)